pypeit/core/flexure.py

```diff
@@ -169,10 +177,27 @@ def spec_flex_shift(obj_skyspec, sky_file, spec_fwhm_pix=None, mxshft=20, excess
 
     msgs.warn("If we use Paranal, cut down on wavelength early on")
 
-    # Load Archival sky spectrum
-    arx_skyspec, arx_fwhm_pix = get_archive_spectrum(sky_file, obj_skyspec=obj_skyspec, spec_fwhm_pix=spec_fwhm_pix)
+    if sky_file is None and arx_skyspec is None:
+        msgs.error("sky_file or arx_skyspec must be provided")
+    elif sky_file is not None and arx_skyspec is not None:
+        msgs.warn("sky_file and arx_skyspec both provided. Using arx_skyspec.")
+        sky_file = None
+
+    # Arxiv sky spectrum
+    if arx_skyspec is None:
```

Comment:
Would it be slightly clearer is this read:
`if sky_file is not None:`
Then, you could move the code from line 199 below to inside this if statement (i.e. to keep the two "modes" all in one place). Explicitly:
```
if sky_file is not None:
    # Load arxiv sky spectrum
    msgs.info("Loading the arxiv sky spectrum and computing its spectral FWHM")
    arx_skyspec, arx_fwhm_pix = get_archive_spectrum(sky_file, obj_skyspec=obj_skyspec, spec_fwhm_pix=spec_fwhm_pix)
    if sky_file != 'model':
        # get gaussian sigma (pixels) for smoothing
        smooth_fwhm_pix = get_fwhm_gauss_smooth(arx_skyspec, obj_skyspec, arx_fwhm_pix, spec_fwhm_pix=spec_fwhm_pix)

        if smooth_fwhm_pix is None:
            # smooth_fwhm_pix is None if spec_fwhm_pix<0, i.e., the wavelength calibration is bad
            msgs.warn('No flexure correction could be computed for this slit/object')
            return None
        if smooth_fwhm_pix > 0:
            arx_skyspec = arx_skyspec.gauss_smooth(smooth_fwhm_pix)
elif arx_fwhm_pix is not None:
    smooth_fwhm_pix = None
    ...
```
Does this capture the same control flow, but groups the modes together?

Reply:
Thanks about the suggestion. It actually allowed me to find a small bug! I re-arrange a bit the flow.